lib/zoho_hub/connection.rb

```diff
@@ -85,32 +86,42 @@ def log(text)
       puts Rainbow("[ZohoHub] #{text}").magenta.bright
     end
 
-    private
+    def refresh_token!
+      was_locked = @mutex.locked?
+      @mutex.synchronize do
+        next if was_locked
```

Comment:
```suggestion
      @mutex.synchronize do
        next unless @mutex.owned?
```

https://ruby-doc.org/core-2.5.0/Mutex.html#method-i-owned-3F

Reply:
@Gnirt  this doesn't seem to work 🤔
```ruby
mutex = Mutex.new
inc = 0
2.times.map do
  Thread.new do
    mutex.synchronize do
      sleep 1
      next unless mutex.owned?
      inc += 1
    end
  end
end.each(&:join)
[31] pry(main)> puts inc
2
=> nil
```

Reply:
Ok 😛  Thanks for trying !